### PR TITLE
fix(core): Copy `packageManager` property when using `generatePackageJson`

### PR DESCRIPTION
**Pull request by stephenwade**
## Current Behavior
When generating `package.json` files during a build (using `generatePackageJson`), the `packageManager` field from the root isn't included.

## Expected Behavior
When generating `package.json` files during a build, the `packageManager` field is copied from the original `package.json` file. This allows us to use Corepack to keep pnpm in sync between dev and prod.

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by FrozenPandaz — DISMISSED** [review]
This makes sense to me. Could you please update the PR so that the PR checks pass please?

**Comment by meeroslav** [comment]
@stephenwade please let us know if you would appreciate help with finalizing this PR.

**Comment by stephenwade** [comment]
Sorry I missed this earlier. Yes, I will take a look at the failing tests today.

**Comment by stephenwade** [comment]
> Too long with no output (exceeded 1h0m0s): context deadline exceeded

Looks like the CI error is a timeout this time, not related to my change.

**Review by meeroslav — APPROVED** [review]

**Comment by stephenwade** [comment]
Sorry, I clicked the wrong button and dismissed @meeroslav's review. I think it's waiting for a re-review from @FrozenPandaz  also.

**Comment by stephenwade** [comment]
@meeroslav @FrozenPandaz Hope this can be merged soon!

File: packages/nx/src/plugins/js/package-json/create-package-json.ts
Comment (JamesHenry) [inline packages/nx/src/plugins/js/package-json/create-package-json.ts:180]: Having it use the root package.json by default makes a lot of sense to me, but I could imagine edge cases where the generated package.json, and the environment in which it is used, do not match the environment of the workspace 1:1.

To allow for users to handle such a case I think it would be make sense to preferentially respect the `packageManager` field on the project package.json (if the file and field exist) and use that when generating.

What do you think @FrozenPandaz?

Comment (meeroslav) [reply packages/nx/src/plugins/js/package-json/create-package-json.ts:180]: Respecting local package.json seems reasonable but also leaves room for user to specify incompatible package manager, which would then mess up the lock file generation.

The root package.json package manager is the one we used for generation and testing and therefore the only one for which we can guarantee things to work.

**Review by meeroslav — APPROVED** [review]

**Review by meeroslav — APPROVED** [review]

**Review by meeroslav — APPROVED** [review]

**Comment by stephenwade** [comment]
The tests are failing on this line: `if (rootPackageJson.packageManager) {`
With the message: `TypeError: Cannot read properties of undefined (reading 'packageManager')`

<details>
```
 FAIL  src/plugins/js/package-json/create-package-json.spec.ts
  ● createPackageJson › parsing "package.json" versions › should use fixed versions when creating package json for apps

    TypeError: Cannot read properties of undefined (reading 'packageManager')

      181 |   );
      182 |
    > 183 |   if (rootPackageJson.packageManager) {
          |                       ^
      184 |     packageJson.packageManager = rootPackageJson.packageManager;
      185 |   }
      186 |

      at createPackageJson (src/plugins/js/package-json/create-package-json.ts:183:23)
      at Object.<anonymous> (src/plugins/js/package-json/create-package-json.spec.ts:506:31)

  ● createPackageJson › parsing "package.json" versions › should override fixed versions with local ranges when creating package json for apps

    TypeError: Cannot read properties of undefined (reading 'packageManager')

      181 |   );
      182 |
    > 183 |   if (rootPackageJson.packageManager) {
          |                       ^
      184 |     packageJson.packageManager = rootPackageJson.packageManager;
      185 |   }
      186 |

      at createPackageJson (src/plugins/js/package-json/create-package-json.ts:183:23)
      at Object.<anonymous> (src/plugins/js/package-json/create-package-json.spec.ts:534:26)

  ● createPackageJson › parsing "package.json" versions › should use range versions when creating package json for libs

    TypeError: Cannot read properties of undefined (reading 'packageManager')

      181 |   );
      182 |
    > 183 |   if (rootPackageJson.packageManager) {
          |                       ^
      184 |     packageJson.packageManager = rootPackageJson.packageManager;
      185 |   }
      186 |

      at createPackageJson (src/plugins/js/package-json/create-package-json.ts:183:23)
      at Object.<anonymous> (src/plugins/js/package-json/create-package-json.spec.ts:555:26)

  ● createPackageJson › parsing "package.json" versions › should override range versions with local ranges when creating package json for libs

    TypeError: Cannot read properties of undefined (reading 'packageManager')

      181 |   );
      182 |
    > 183 |   if (rootPackageJson.packageManager) {
          |                       ^
      184 |     packageJson.packageManager = rootPackageJson.packageManager;
      185 |   }
      186 |

      at createPackageJson (src/plugins/js/package-json/create-package-json.ts:183:23)
      at Object.<anonymous> (src/plugins/js/package-json/create-package-json.spec.ts:586:26)
```
</details>

I think `rootPackageJson` should be defined here. Several other places in this file reference it as if it is always defined, so I don't understand why it would be undefined in this test.

**Pull request closed without merging**
